contracts/contracts/strategies/BridgedWOETHStrategy.sol

```diff
@@ -0,0 +1,303 @@
+
+    IWETH9 public immutable weth;
+    IERC20 public immutable bridgedWOETH;
+    IERC20 public immutable oethb;
+
+    uint256 public constant MAX_PRICE_STALENESS = 2 days;
+
+    uint128 public lastOraclePrice;
+    uint128 public maxPriceDiffBps;
+
+    /**
+     * @dev Verifies that the caller is the Governor or Strategist.
+     */
+    modifier onlyGovernorOrStrategist() {
+        require(
+            isGovernor() || msg.sender == IVault(vaultAddress).strategistAddr(),
+            "Caller is not the Strategist or Governor"
+        );
+        _;
+    }
+
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        address _weth,
+        address _bridgedWOETH,
+        address _oethb
+    ) InitializableAbstractStrategy(_stratConfig) {
+        weth = IWETH9(_weth);
+        bridgedWOETH = IERC20(_bridgedWOETH);
+        oethb = IERC20(_oethb);
+    }
+
+    function initialize(uint128 _maxPriceDiffBps)
+        external
+        onlyGovernor
+        initializer
+    {
+        InitializableAbstractStrategy._initialize(
+            new address[](0), // No reward tokens
+            new address[](0), // No assets
+            new address[](0) // No pTokens
+        );
+
+        _setMaxPriceDiffBps(_maxPriceDiffBps);
+    }
+
+    /**
+     * @dev Sets the max price diff bps for the wOETH value appreciation
+     * @param _maxPriceDiffBps Bps value, 10k == 100%
+     */
+    function _setMaxPriceDiffBps(uint128 _maxPriceDiffBps) internal {
+        require(
+            _maxPriceDiffBps > 0 && _maxPriceDiffBps <= 10000,
+            "Invalid bps value"
+        );
+
+        emit MaxPriceDiffBpsUpdated(maxPriceDiffBps, _maxPriceDiffBps);
+
+        maxPriceDiffBps = _maxPriceDiffBps;
+    }
+
+    /**
+     * @dev Sets the max price diff bps for the wOETH value appreciation
+     * @param _maxPriceDiffBps Bps value, 10k == 100%
+     */
+    function setMaxPriceDiffBps(uint128 _maxPriceDiffBps)
+        external
+        onlyGovernor
+    {
+        _setMaxPriceDiffBps(_maxPriceDiffBps);
+    }
+
+    /**
+     * @dev Finds the value of bridged wOETH from the Oracle.
+     *      Ensures that it's within the bounds and reasonable.
+     *      And stores it.
+     *
+     *      NOTE: Intentionally not caching `Vault.priceProvider` here,
+     *      since doing so would mean that we also have to update this
+     *      strategy every time there's a change in oracle router.
+     *      Besides on L2, the gas is considerably cheaper than mainnet.
+     *
+     * @return Latest price from oracle
+     */
+    function _updateWOETHOraclePrice() internal returns (uint256) {
+        // WETH price per unit of bridged wOETH
+        uint256 oraclePrice = IOracle(IVault(vaultAddress).priceProvider())
+            .price(address(bridgedWOETH));
+
+        // 1 wOETH > 1 WETH, always
+        require(oraclePrice > 1 ether, "Invalid wOETH value");
+
+        uint128 oraclePrice128 = oraclePrice.toUint128();
```

Comment:
Can we just convert  the lastOraclePrice to uint256 here, and then we don't have to do any more casting in the rest of the function, outside of a storage set? Raw casts in the middle of math scare me.

Reply:
We can change the events to uint256s as well.

Reply:
Or just do whatever you want on this one. :D It's less straightforward than I thought at first.

Reply:
This is one messy function, yeah. I removed one unnecessary cast. Let me see if I can make it less messy